odoo/addons/base/models/res_bank.py

```diff
@@ -108,6 +108,10 @@ def retrieve_acc_type(self, acc_number):
         """ To be overridden by subclasses in order to support other account_types.
         """
         return 'bank'
+    
+    def name_get(self):
+        return [(acc.id, '{} - {}'.format(acc.acc_number, acc.bank_id.display_name) if acc.bank_id else acc.acc_number)
```

Comment:
The display_name of res.bank also contains the bic, which is not really useful to show here and makes the text longer and less readable. Instead, use acc.bank_id.name.